libs/glob/Match.ml

```diff
@@ -134,19 +138,24 @@ let compile ~source pat =
   { source; re }
 [@@profiling "Glob.Match.compile"]
 
-(* This is used during unit testing. *)
-let debug = ref false
-
 let run matcher path =
   let res = Pcre_.pmatch_noerr ~rex:matcher.re path in
-  if !debug then
-    (* expensive string concatenation; may not be suitable for logger#debug *)
-    Printf.eprintf "** glob: %S  pcre: %s  path: %S  matches: %B\n%!"
-      matcher.source.line_contents matcher.re.pattern path res;
+  (* perf: this gets called a lot. The match-with is expected to make things
+     faster by creating a closure for the anonymous function only in debug
+     mode. *)
+  (match Logs.level () with
+  | Some Debug ->
+      Logs.debug (fun m ->
+          m ~tags "glob: %S  pcre: %s  path: %S  matches: %B"
+            matcher.source.line_contents matcher.re.pattern path res)
+  | _ -> ());
```

Comment:
This makes `--debug` very noisy. Maybe having a way to not log this unless explicitly selected by a tag would be good?